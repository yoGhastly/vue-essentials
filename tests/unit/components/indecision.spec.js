import { shallowMount } from "@vue/test-utils";
import Indecision from "@/components/indecision";

describe("Indecision vue", () => {
  let wrapper, clgSpy;
  beforeEach(() => {
    wrapper = shallowMount(Indecision);
    clgSpy = jest.spyOn(console, "log");
  });

  test("debe hacer match con el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("escribir en el input no debe de disparar nada", async () => {
    const getAnswerSpy = jest.spyOn(wrapper.vm, "getAnswer");
    const input = wrapper.find("input");
    await input.setValue("Hola mundo");
    expect(clgSpy).toHaveBeenCalledTimes(1);
    expect(getAnswerSpy).not.toHaveBeenCalled();
  });

  test("escribir el simobolo de '?' debe disparar el fetch", () => { });

  test("pruebas en getAnswer", () => { });

  test("pruebas en getAnswer - fallo en API", () => { });
});
