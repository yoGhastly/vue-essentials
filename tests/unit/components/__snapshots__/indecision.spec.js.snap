// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indecision vue debe hacer match con el snapshot 1`] = `
<img alt="Placeholder">
<div class="bg-dark"></div>
<div class="indecision-container"><input type="text" placeholder="Hazme una pregunta">
  <p>Recuerda terminar con un signo de interrogacion (?)</p>
  <!--v-if-->
</div>
`;
